python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,559 @@
+            self.set_index_name(index)
+            self.mod = mod
+            self.input_bindings = PipelineConfig.BindingList(self, "input")
+            self.output_bindings = PipelineConfig.BindingList(self, "output")
+            self.target_host_ = None
+            self.build_func_ = None
+            self.params_ = None
+            self.target_ = None
+            self.dev_ = None
+
+        def __getitem__(self, key):
+            """get item by key"""
+            if isinstance(key, str):
+                if key == "input":
+                    return self.input_bindings
+
+                if key == "output":
+                    return self.output_bindings
+
+            raise RuntimeError(f"{key} not found!")
+
+        def get_data_type(self, key, stype):
+            """get module input/output data type."""
+
+            if stype == "input":
+                for param in self.input_params:
+                    if param.name_hint == key:
+                        return param._checked_type_
+
+            if stype == "output":
+                if isinstance(self.output_values, tvm.ir.type.TupleType):
+                    if int(key) < len(self.output_values.fields):
+                        return self.output_values.fields[int(key)]
+                elif int(key) == 0:
+                    return self.output_values
+
+            return None
+
+        def set_index_name(self, index):
+            """generate name by index and storage index value"""
+            self.index = index
+            self.name = "mod{}".format(str(index))
+
+        def is_root_mod(self):
+            """use by DAG topology sort, identify if this item is root item and in-degree is 0"""
+            for _, binding in self.input_bindings.bindings.items():
+                if binding.parents:
+                    return False
+
+            return True
+
+        def remove_self_from_bindings(self):
+            """use by DAG topology sort, by remove self from binding to reduce child in-degree"""
+            for _, binding in self.output_bindings.bindings.items():
+                for child in binding.bindings:
+                    if binding in child.parents:
+                        child.parents.remove(binding)
+
+        def set_target_host(self, host):
+            """set target host that use by build function"""
+            self.target_host_ = host
+
+        def set_build_func(self, build_func):
+            """set build funciton that use by build function"""
+            self.build_func_ = build_func
+
+        def set_params(self, params):
+            """set params that use by build function"""
+            self.params_ = params
+
+        def set_target(self, target):
+            """set target that use by build function"""
+            self.target_ = target
+
+        def set_dev(self, dev):
+            """set dev that use by build function"""
+            self.dev_ = dev
```

Comment:
As I mentioned, Python doesn't have private member, so you can just name and access them normally. For example, `mod.set_target_host(host)` -> `mod.target_host = host`.

Reply:
fixed.